lib/pbench/server/api/resources/query_apis/metadata_index/iteration_samples.py

```diff
@@ -0,0 +1,356 @@
+            )
+
+
+class IterationSamplesRows(RunIdBase):
+    """
+    Iteration samples API that returns iteration sample rows after
+    applying client specified filters.
+    """
+
+    DOCUMENT_SIZE = 10000  # Number of documents to return in one page
+    SCROLL_EXPIRY = "1m"  # Scroll id expires in 1 minute
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("filters", ParamType.JSON, required=False),
+                Parameter("run_id", ParamType.STRING, required=True),
+                Parameter("scroll_id", ParamType.STRING, required=False),
+            ),
+        )
+
+    def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
+        """
+        Construct a pbench Elasticsearch query for filtering iteration
+        samples based on a given run id and other filtering parameters
+        specified in JSON payload.
+
+        Note: If the ES scroll id is present we will ignore the filters
+        parameter and instead construct an Elasticsearch query for scrolling
+        based on a client provided scroll id.
+
+        If a scroll_id is specified, the query will return the next page of
+        results of the original query; otherwise we form a new query, either
+        with the specified filters or without any filters (returning up to 10000
+        documents at a time).
+
+        Args:
+            json_data:
+                "run_id": run id of a dataset that client wants to search for
+                        fetching the iteration samples.
+
+                "scroll_id": Server provided Elasticsearch scroll id that client
+                            recieved in the result of the original query with
+                            filters. This can be used to fetch the next page of
+                            the result.
+
+                "filters": key-value representation of query filter parameters to
+                           narrow the search results e.g. {"sample.name": "sample1"}
+
+        EXAMPLES:
+            {
+                "run_id": "1234567"
+                "filters": {"sample.name": "sample1"},
+            }
+            or
+            {
+                "run_id": "1234567"
+                "scroll_id": "cmFuZG9tX3Njcm9sbF9pZF9zdHJpbmdfMg=="
+            }
+        """
+        run_id = context["run_id"]
+        dataset = context["dataset"]
+        scroll_id = json_data.get("scroll_id")
+
+        self.logger.info(
+            "Return iteration sample rows {} for dataset {}, prefix {}, run id {}",
+            "next page " if scroll_id else "",
+            dataset,
+            self.prefix,
+            run_id,
+        )
+
+        scroll_id = json_data.get("scroll_id")
+        if scroll_id:
+            return {
+                "path": "/_search/scroll",
+                "kwargs": {
+                    "json": {
```

Comment:
Wait; shouldn't this be `"params"` rather than `"json"`??

Reply:
No, those "parameters" are supposed to conveyed in the payload.  (I think you can pass the ID as a parameter, but I believe that that practice is deprecated.)

Reply:
Because, at the moment, each of these will do the same thing:
```
GET /_search/scroll/<scroll_id>?scroll="1m"
GET /_search/scroll?scroll_id=<scroll_id>&scroll="1m"
POST /_search/scroll/<scroll_id>?scroll="1m"
POST /_search/scroll?scroll_id=<scroll_id>&scroll="1m"
POST /_search/scroll
    BODY { "scroll_id"=<scroll_id>, "scroll"="1m"}
```
but four of them are now deprecated.